usdoj-crt/crt-portal
# Day 1 of spike on internationalization with @el-mapache

[Link to ZenHub issue.](https://app.zenhub.com/workspaces/doj-crt-intake-scrum-board-5d03bf56c1c8a35d482eca0f/issues/18f/crt-portal/152)

## What does this change?

This is the work product of the first day of a 2-day `i18n` spike with @el-mapache.

In this PR, we got the fundamentals of the Django i18n system working — defining a translation file with a few fake-translated strings, and getting Django to surface those fake translations to a user.

### What's next?

There's still more to do — next, we'll work on telling Django to translate strings that are stored in the JavaScript code and in the database. And of course setting a user's language preference (through a dropdown menu or browser preferences) is a separate topic. As much as we get done in this 2-day time-boxed spike, we get done.

## How to test this PR

1. In `settings.py`, change the value of `LANGUAGE_CODE` from `en-us` to `es`.
2. Navigate to the `es` `django.po` file:  `crt_portal/cts_forms/locale/es/LC_MESSAGES/django.po`.
3. Change the value of `msgstr` for one of the strings you'd like to translate — supply any test value you like (pseudolocalization testing). 
4. Run `docker-compose run web django-admin compilemessages`.
5. Start the site locally and navigate to a page that contains the translated content to see your fake test translation value. 

## Screenshots (for front-end PR):

Fake pseudolocalization for "Next" button: 

<img width="819" alt="translation-screenshot" src="https://user-images.githubusercontent.com/3209501/69353691-7fd20080-0c44-11ea-89e6-be81e464ac0f.png">

## Checklist:

### Author

+ [x] Check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).
